#summary ApicoAP (version 2) User Manual

== Introduction ==

ApicoAP is a program for predicting Apicoplast targeted proteins that use bipartite signaling mechanism (ApicoTPs). It comes with both graphical user and command line interfaces.

== Instructions for running ApicoAP ==

* Requirements:* Python version 2.7x is needed to launch the software, which can be downloaded from http://www.python.org/download/releases/2.7.3/ . First, unpack the archive ApicoAPxx.zip on your path of choice. Please note some files in this directory are essential and any damage to them would cause ApicoAP stop working. 


*NOTE:* Sometimes, run-time messages (mostly warnings) will occur due to different versions of some of the supporting software. These messages should be ignored since they are benign.

* Launching ApicoAP with Graphical User Interface (GUI): *

  * For Mac OS X, simply double-clicking ApicoAP_GUI.py would launch the software if the default program to open ".py" files is set to be the terminal. If instead of the program, an editor opens the source code, and you don't know how to change the default program to open a file, then, open the Terminal app, change directories to the location of ApicoAP, and type:
      python ApicoAP_GUI.py
	
  * _ NOTE on MAC OS X: _ Splash screen of the program may not appear when above instructions were followed. Instructions will be included as a separate wiki page for you to convert .py file to an .app file at the end of this document. When app file is launched splash screen will be visible. 
	
  * For Windows, simply double click on ApicoAP_GUI.py or open a command prompt, change directories to the location of ApicoAP, and type:  
      python ApicoAP_GUI.py
	
* Launching ApicoAP from command line: * Please refer to the section entitled "ApicoAP - command line"

== Instructions for using ApicoAP GUI == 

  * Select the organism you are interested in. The classification rule that applies to the proteins of the selected organism will be used in the prediction.

  * Copy-paste FASTA formatted protein sequences to the text box, or import a FASTA formatted file by selecting the "Import sequences from fasta file" option. *NOTE:* exampleFiles directory contains some FASTA formatted files that can be used for testing ApicoAP. 

  * Click "Run ApicoAP" button to see the predictions along with id and SP cleavage site information.

  * Optional: Save predictions to a file.

  * Click "Enter new data" button to enter new data. *Note:* Clicking this button enables the use of the rest of the form that is disabled after prediction results are displayed.


*NOTE:* For the prediction results to be displayed as neatly as possible, description lines of the fasta formatted sequences should be in an acceptable format. Following exemplifies acceptable description line formats for the same sequence. (first two are standards used in eupathdb and ncbi):

{{{
>PF3D7_0815700  | Plasmodium falciparum 3D7 | ubiquitin (Ub) | protein  | length=373
MIKKVHICLFIIFFYVIFLIHICKGIRLQNYKNERINNRHMLNTIRNNVSIYQNKHISNN
NTKENKCNIMINYYDKSNIFCKSFLLSMEEKDNIKNIKKKIEEIYGIPLTLQEILYDNKK
LENNITIQNIIKDKQIKILNLRLITILPHLFLQKDDDNNTNKRNDVSSSSSSSSLYNNEY
IKSNKRITYLKNKLTYYGYLTLLNEYKKLSHILENKKYILKNDDILESFKSFDQEFEKTL
KNNNINLHKIKKEINKLKHIDKKKLLLRLEVDYPLMSNNLTKRIKQLFQYYYMGDITTII
KFSIFFYILYKYADYPNHIKKFFLYLSILFLISPFKPFYKFSHFLFFSVPNNILFSGFTN
ILSASYQQILMCQ
}}}
{{{
>gi|296005015|ref|XP_002808847.1| conserved Plasmodium protein, unknown function [Plasmodium falciparum 3D7]
MIKKVHICLFIIFFYVIFLIHICKGIRLQNYKNERINNRHMLNTIRNNVSIYQNKHISNN
NTKENKCNIMINYYDKSNIFCKSFLLSMEEKDNIKNIKKKIEEIYGIPLTLQEILYDNKK
LENNITIQNIIKDKQIKILNLRLITILPHLFLQKDDDNNTNKRNDVSSSSSSSSLYNNEY
IKSNKRITYLKNKLTYYGYLTLLNEYKKLSHILENKKYILKNDDILESFKSFDQEFEKTL
KNNNINLHKIKKEINKLKHIDKKKLLLRLEVDYPLMSNNLTKRIKQLFQYYYMGDITTII
KFSIFFYILYKYADYPNHIKKFFLYLSILFLISPFKPFYKFSHFLFFSVPNNILFSGFTN
ILSASYQQILMCQ
}}}
{{{
>296005015
MIKKVHICLFIIFFYVIFLIHICKGIRLQNYKNERINNRHMLNTIRNNVSIYQNKHISNN
NTKENKCNIMINYYDKSNIFCKSFLLSMEEKDNIKNIKKKIEEIYGIPLTLQEILYDNKK
LENNITIQNIIKDKQIKILNLRLITILPHLFLQKDDDNNTNKRNDVSSSSSSSSLYNNEY
IKSNKRITYLKNKLTYYGYLTLLNEYKKLSHILENKKYILKNDDILESFKSFDQEFEKTL
KNNNINLHKIKKEINKLKHIDKKKLLLRLEVDYPLMSNNLTKRIKQLFQYYYMGDITTII
KFSIFFYILYKYADYPNHIKKFFLYLSILFLISPFKPFYKFSHFLFFSVPNNILFSGFTN
ILSASYQQILMCQ
}}}

== Interpreting prediction results ==

Prediction for a protein can be one of the following: "non-ApicoTP", "ApicoTP", "No SP", "Unknown id". First two tags are self-explanatory. "No SP" implies that the id information is tied with the information that the protein does not contain an SP region. Since ApicoAP predicts Apicoplast Targeted Proteins (ApicoTPs)that use bipartite signaling mechanism, which dictates a protein to contain an SP region; this prediction implies that the protein is not an ApicoTP. 


If the program cannot tie the given ID with an SP cleavage information, prediction 
result for the protein will be "Unknown id". User is encouraged to enter SP cleavage site information for these proteins by either preparing and importing a custom file or entering this information manually (follow use cases B and C for more information).
 
== Preparing/Entering SP cleavage site information ==

When entering SP cleavage site information manually (or through a custom prepared file), one should put the end index of the SP region as the SP cleavage site assuming the beginning index is 1.

SP cleavage site information file is expected to contain id-SP cleavage site pairs in separate lines. Each line should contain an ID and SP cleavage site (either a number or '-' or 'N' implying no SP), seperated by a tab. 


== ApicoAP - command line ==

{{{
*Usage:* python ApicoAP_CL.py -m ORGANISM [-s SP_FILE] -f FASTA_FILE -o OUT_FILE
*Options:*
  -h, --help
           show this help message and exit
  -m ORGANISM, --mode=ORGANISM   
           Select organism: BBOV for B.bovis, PF for P.falciparum, 
           TOXO for T.gondii, PY for P.yoelii.[default: BBOV]
  -s FILE, --spFile=FILE
           Read additional SP cleavage site information from FILE
  -f FILE, --fastaFile=FILE
           Read FASTA formatted sequences from FILE
  -o FILE, --outputFile=FILE
           Write output to FILE
}}}                        
All options but "-s" are required. When additional SP cleavage site information is provided by this option, provided file is imported "after" the default files are imported (which are the files named as "organism".spCleavs and "organism".idMap). This is important since whatever is imported later overwrites the SP cleavage site information for the already existing protein IDs.

----

* Reference to the ApicoAP paper (where you can read more about the method): *

ApicoAP: The first computational model for identifying apicoplast-targeted proteins in multiple species of Apicomplexa

Gokcen Cilingir1, Shira L. Broschat1,2, and Audrey O.T. Lau2

1School of Electrical Engineering and Computer Science, Washington State University, Pullman, WA 99164, USA

2Program of Genomics, Department of Veterinary Microbiology and Pathology, Paul G. Allen School for Global Animal Health, Washington State University, Pullman, WA 99164, USA